### ISSUE TITLE
need to return a drawing path from isolate

### ISSUE THREAD

**Issue by premingiet** [dart-lang/language#1332]
As you know isolate works by sending messages back and forth between isolates, and for that only string type can be passed or fetched from isolate. Basically if i need to work on int or double type of data, then i need to send that as string and then later parse it as primitive type. 

But what if i want to work on Offsets and then make a drawing path out of all the offsets and then finally wish to return that drawing path ? I am not sure if it is even possible to do inside an isolate. Because i have no idea how to return a path as a string message and then later parse it back as drawing path. I hope it makes sense what i just mentioned.

**Comment by mateusfccp** [comment]
Isolates don't work only with `String`. From the [documentation](https://api.dart.dev/stable/2.10.4/dart-isolate/SendPort/send.html):

> The content of message can be: primitive values (null, num, bool, double, String), instances of SendPort, and lists and maps whose elements are any of these. List and maps are also allowed to be cyclic.

So I also already sent messages with custom objects composed of these primitive types, but there's not in the documentation specifying when it is possible. Functions definitely can't be passed through isolates.

**Comment by premingiet** [comment]
yes what i actually meant was all the supported types for isolate message passing are "null, num, bool, double, String" but as you can see all those types can surely be converted to string and vice versa. But consider you wish to pass a `dart:ui path` and as you know path is a complex, one-dimensional object. Paths consist of lines, arcs, beziers and so on and so forth.

So if i pass bunch of offsets as string json message, then in the isolate i wish to do my heavy task of complex manipulation of those offsets and generate a path from it. Then when that process is over, i wish to send that path back to main ui where i can render that path in the paint method. Now main issue is i can't send the path back to main ui.
